src/ucp/core/ucp_request.c

```diff
@@ -290,8 +290,11 @@ ucs_status_t ucp_request_test(void *request, ucp_tag_recv_info_t *info)
     ucp_request_t *req = (ucp_request_t*)request - 1;
 
     if (req->flags & UCP_REQUEST_FLAG_COMPLETED) {
+        /* NOTE: workaround for gtest */
```

Comment:
why not fix gtest instead?
ucp_request_test() should not be used by gtest as well, since it's deprecated

Reply:
I reuse common function `wait` adding return value `length` in next PR to count how many of data received.

Reply:
the lib code should not be adapted to the test, it's the other way around